fe/fe-core/src/main/java/com/starrocks/catalog/S3TableFunctionTable.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    @Override
+    public List<Column> getFullSchema() {
+        if (fullSchema.isEmpty()) {
+            fullSchema.add(new Column("name", Type.VARCHAR));
+            fullSchema.add(new Column("age", Type.INT));
+
+            return fullSchema;
+        }
+
+        TNetworkAddress address = new TNetworkAddress();
+        List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().getBackendIds(true);
+        if (backendIds.isEmpty()) {
+            return null;
+        }
+
+        Collections.shuffle(backendIds);
+        Backend be = GlobalStateMgr.getCurrentSystemInfo().getBackend(backendIds.get(0));
+        address = new TNetworkAddress(be.getHost(), be.getBrpcPort());
+        if (address == null) {
+            return null;
+        }
+
+        try {
+            BackendServiceClient.getInstance().getFileSchema(address, new PFileSchemaRequest());
+        } catch (RpcException e) {
+            return null;
+        }
+
+        return fullSchema;
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of file table, " + "they are path and format");
+        }
+
+        String path = properties.get(JSON_KEY_FILE_PATH);
+        if (Strings.isNullOrEmpty(path)) {
+            throw new DdlException("path is null. Please add properties(path='xxx') when create table");
+        }
+
+        String format = properties.get(JSON_KEY_FORMAT);
+        if (Strings.isNullOrEmpty(format)) {
+            throw new DdlException("format is null. Please add properties(format='xxx') when create table");
+        }
+        if (!format.equalsIgnoreCase("parquet") && !format.equalsIgnoreCase("orc")) {
+            throw new DdlException("not supported format: " + format);
+        }
+    }
+
+    public String getTableLocation() {
+        return fileProperties.get(JSON_KEY_FILE_PATH);
+    }
+
+    public RemoteFileInputFormat getFileFormat() {
+        if (fileProperties.get(JSON_KEY_FORMAT).equalsIgnoreCase("parquet")) {
+            return RemoteFileInputFormat.PARQUET;
+        } else if (fileProperties.get(JSON_KEY_FORMAT).equalsIgnoreCase("orc")) {
+            return RemoteFileInputFormat.ORC;
+        } else {
+            return RemoteFileInputFormat.UNKNOWN;
+        }
+    }
+
+    public Map<String, String> getFileProperties() {
+        return fileProperties;
+    }
+
+    public List<RemoteFileDesc> getFileDescs() throws DdlException {
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(fileProperties, null);
+        Configuration configuration = hdfsEnvironment.getConfiguration();
+        HiveRemoteFileIO remoteFileIO = new HiveRemoteFileIO(configuration);
+        boolean recursive = Boolean.parseBoolean(fileProperties.getOrDefault(JSON_RECURSIVE_DIRECTORIES, "false"));
+        RemotePathKey pathKey = new RemotePathKey(getTableLocation(), recursive, Optional.empty());
```

Comment:
Whether to support multiple paths like broker load?

Reply:
Already updated.